nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -1,60 +1,115 @@
 
-    # Electrodes columns descriptions
-    metadata["Ecephys"]["Electrodes"] = [
-        dict(name="shank_electrode_number", description="0-indexed channel within a shank."),
-        dict(name="shank_group_name", description="The name of the ElectrodeGroup this electrode is a part of."),
-    ]
+    return dict_deep_update(metadata, metadata_update)
+
+
+def _get_source_schema(cls):
+    source_schema = get_schema_from_method_signature(
+        class_method=cls.__init__, exclude=["x_pitch", "y_pitch", "stream_id"]
+    )
+    source_schema["properties"]["folder_path"]["description"] = (
+        "Path to folder containing SpikeGLX files " "(using spikeinterface)."
+    )
+    source_schema["properties"]["file_path"]["description"] = "Path to SpikeGLX file (using spikeextractors)."
+    return source_schema
 
 
 class SpikeGLXRecordingInterface(BaseRecordingExtractorInterface):
     """Primary data interface class for converting the high-pass (ap) SpikeGLX format."""
 
-    RX = SpikeGLXRecordingExtractor
+    RX = SpikeGLXRecordingExtractorSI
 
     @classmethod
     def get_source_schema(cls):
-        source_schema = get_schema_from_method_signature(class_method=cls.__init__, exclude=["x_pitch", "y_pitch"])
-        source_schema["properties"]["file_path"]["description"] = "Path to SpikeGLX file."
-        return source_schema
+        return _get_source_schema(cls)
+
+    def __init__(
+        self,
+        folder_path: FilePathType = None,
+        file_path: FilePathType = None,
+        spikeextractors_backend: Optional[bool] = False,
+        stub_test: Optional[bool] = False,
+        **kwargs,
+    ):
+        """
+        Load and prepare raw acquisition data and corresponding metadata from the Neuropixels format.
+
+        Parameters
+        ----------
+        folder_path: PathType
+            folder containing pair of .ap/.meta or .lf/.meta file
+        file_path: PathType
+            path to .ap/.lf file if using spikeextractors_backend
+        spikeextractors_backend : Optional[bool], optional
+            False by default. When True the interface uses the old extractor from the spikextractors library instead
+            of a new spikeinterface object.
+        stub_test: bool
+        kwargs
+            additional args depending on usage of spikeextractors or spikeinterface
+        """
+        if spikeextractors_backend:
+            assert file_path is not None and Path(file_path).is_file(), f"{file_path} should be a file for version='v1'"
+            self.RX = se.SpikeGLXRecordingExtractor
+            super().__init__(file_path=str(file_path), **kwargs)
+            self.recording_extractor = OldToNewRecording(oldapi_recording_extractor=self.recording_extractor)
+            self.folder_path = file_path.parent
+        else:
+            assert (
+                folder_path is not None and Path(folder_path).is_dir()
+            ), f"{folder_path} should be a folder for version='v2'"
```

Comment:
The string here should reference the backend instead of the version, correct?

Reply:
yes, thanks, have changed that